pkg/agent/protocol/metricmap_test.go

```diff
@@ -9,18 +9,34 @@ import (
 func TestMetricMap(t *testing.T) {
 	t.Parallel()
 
+	t.Run("initializes metrics", func(t *testing.T) {
+		t.Parallel()
+
+		const foo = "foo_metric"
+
+		mm := protocol.NewMetricMap(foo)
+		fooMetric, hasFoo := mm.Map()[foo]
+		if !hasFoo {
+			t.Fatalf("foo should be explicitly set")
```

Comment:
Minor thing: I find this message confusing because foo was explicitly set. Maybe "foo should exist" makes clear why it failed.